src/app/icd/BUILD.gn

```diff
@@ -34,6 +34,8 @@ source_set("notifier") {
 # All sources and configurations used by the ICDManager need to go in this source-set
 source_set("manager") {
   sources = [
+    "ICDCheckInSender.cpp",
+    "ICDCheckInSender.h",
```

Comment:
can you brake this out of the source set and have another one for the check-in sender?

Reply:
Nope, ICDManager will have to know about the ICD sender in a follow up PR

Reply:
So add it as a dependency like what was done for all the other source sets?

Reply:
But why should the ICD  have more than 1 sources set ? When in fact these are required for the feature ?